### ISSUE TITLE
Using dropout in Timedistributed

### ISSUE THREAD

**Issue by tivaro** [fchollet/keras#3995]
In my complex model (build using the functional API), I have a dense layer, that I want to repeat over the middle dimension of a 3D input. The `TimeDistributed` layer is perfect for this task (funny thing is that in my case it has nothing to do with time ;), but still).

I want to use dropout, but the dropout will also have to be distributed over the middle dimension. I think the correct behaviour would be that the same datapoints are 'dropped out' at each 'timestep'. (I think it wouldn't make sense to have the same weight at each timestep, but random dropout for that weight).

I came up with the following implementation:

```python
data = np.random.rand(2, 5, 3)

input_layer = Input(name='atom_inputs', shape=(5, 3))
output1 = TimeDistributed(Dropout(0.2))(input_layer)
output2 = TimeDistributed(Dense(2))(output1)
model = models.Model(input=[input_layer], output=[output2])
```

Does dropout here actually behave the way I intended?
I tried to investigate this with the above toy example using theano functions (as in [here](https://github.com/fchollet/keras/issues/431#issuecomment-124175958)).

Unfortunately, I cannot get the output at training time, because `TimeDistributed` doesn't have a `get_input` method. I want to make sure to get the output at training time, so I do not know if there are other functions that I could use for that.

**Comment by farizrahman4u** [comment]
```python
input = Input(name='atom_inputs', shape=(5, 3))
ones = Lambda(lambda x: (x * 0 + 1)[:, 0, :], output_shape=lambda s: (s[0], s[2]))
dropped = Dropout(0.2)(ones)
dropped = RepeatVector(5)(dropped)
output1 = Lambda(lambda x: x[0] * x[1], output_shape=lambda s: s[0])([input, dropped])
output2 = TimeDistributed(Dense(2))(output1)
model = models.Model(input=[input], output=[output2])
```

**Comment by tivaro** [comment]
Thanks for your solution!

So is it indeed the case then that `TimeDistributed(Dropout(0.2))(input_layer)` would yield a similar result as `Dropout(0.2)(input_layer)`?

**Comment by farizrahman4u** [comment]
yes.

**Comment by xiangyanchao** [comment]
is SpatialDropout1D that you need?
